tools/cli/commands.go

```diff
@@ -0,0 +1,871 @@
+	if err != nil {
+		ExitIfError(err)
+	}
+	return domainClient
+}
+
+func getWorkflowClient(c *cli.Context) client.Client {
+	domain := getRequiredGlobalOption(c, FlagDomain)
+
+	builder := getBuilder(c)
+	builder.SetDomain(domain)
+	wfClient, err := builder.BuildCadenceClient()
+	if err != nil {
+		ExitIfError(err)
+	}
+
+	return wfClient
+}
+
+func getWorkflowServiceClient(c *cli.Context) workflowserviceclient.Interface {
+	builder := getBuilder(c)
+	client, err := builder.BuildServiceClient()
+	if err != nil {
+		ExitIfError(err)
+	}
+
+	return client
+}
+
+func getRequiredOption(c *cli.Context, optionName string) string {
+	value := c.String(optionName)
+	if len(value) == 0 {
+		ExitIfError(fmt.Errorf("%s is required", optionName))
+	}
+	return value
+}
+
+func getRequiredGlobalOption(c *cli.Context, optionName string) string {
+	value := c.GlobalString(optionName)
+	if len(value) == 0 {
+		ExitIfError(fmt.Errorf("%s is required", optionName))
+	}
+	return value
+}
+
+func getBuilder(c *cli.Context) WorkflowClientBuilderInterface {
+	address := c.GlobalString(FlagAddress)
+	env := c.GlobalString(FlagEnv)
+	customized := c.GlobalBool(FlagCustomizeBuilder)
+
+	var builder WorkflowClientBuilderInterface
+	if customized {
+		// Caller should already call SetBuilder
+		if cBuilder == nil {
+			panic("Customized builder should be set first")
+		}
+		builder = cBuilder
+		builder.SetEnv(ConvertEnv(env)).SetHostPort(address)
+	} else {
+		builder = getLocalBuilder(address)
+	}
+	return builder
+}
+
+func getLocalBuilder(address string) WorkflowClientBuilderInterface {
+	addr := localHostPort
+	if address != "" {
+		addr = address
+	}
+	logger, err := zap.NewDevelopment()
+	if err != nil {
+		panic(err)
+	}
+	builder := NewBuilder(logger).SetHostPort(addr)
+
+	return builder
+}
+
+func convertTime(unixNano int64) string {
+	t2 := time.Unix(0, unixNano)
+	return t2.Format(defaultTimeFormat)
+}
+
+func parseTime(timeStr string, defaultValue int64) int64 {
+	if len(timeStr) == 0 {
+		return defaultValue
+	}
+
+	// try to parse
+	parsedTime, err := time.Parse(defaultTimeFormat, timeStr)
+	if err == nil {
+		return parsedTime.UnixNano()
+	}
+
+	// treat as raw time
+	resultValue, err := strconv.ParseInt(timeStr, 10, 64)
+	if err != nil {
+		ExitIfError(fmt.Errorf("cannot parse time '%s', use UTC format '2006-01-02T15:04:05Z' or raw UnixNano directly. Error: %v", timeStr, err))
+	}
+
+	return resultValue
+}
+
+func getPtrOrNilIfEmpty(value string) *string {
+	if value == "" {
+		return nil
+	}
+	return common.StringPtr(value)
+}
+
+func getCliIdentity() string {
+	hostName, err := os.Hostname()
+	if err != nil {
+		hostName = "UnKnown"
+	}
+	return fmt.Sprintf("cadence-cli@%s", hostName)
+}
+
+func newContext() (context.Context, context.CancelFunc) {
+	return context.WithTimeout(context.Background(), time.Second*5)
+}
+
+func newContextForLongPoll() (context.Context, context.CancelFunc) {
+	return context.WithTimeout(context.Background(), time.Minute*2)
```

Comment:
Do we need to let user specify timeout for this? (Make it optional, and default to 2 minutes.) For example, what if the workflow takes longer than 2 minutes?

Reply:
Yes, changed.